person_capture/gui_app.py

```
@@ -4697,121 +5160,38 @@ def save_hit(c, idx):
-                            pass
-                        cx1, cy1, cx2, cy2 = rx1 + off_x, ry1 + off_y, rx2 + off_x, ry2 + off_y
+                    # Compose the saved dataset crop from identity evidence.
+                    # The detector/person/face boxes identify the target; they do not
+                    # directly dictate the final crop.  The composition stage chooses
+                    # close/portrait/body framing while preserving the invariant that
+                    # the target face/head or person is not cut off when a feasible
+                    # in-bounds ratio crop exists.
+                    if bool(getattr(cfg, "auto_crop_borders", False)):
+                        bx1 = int(locals().get("off_x", 0))
+                        by1 = int(locals().get("off_y", 0))
+                        bx2 = int(bx1 + locals().get("W2", W))
```

Ruff (0.15.11)

[warning] Function definition does not bind loop variable `W`

(B023)